aten/src/ATen/native/native_functions.yaml

```diff
@@ -3728,7 +3728,7 @@
     CPU: prelu_backward_cpu
     CUDA: prelu_backward_cuda
 
-- func: gelu.out(Tensor self, *, Tensor(a!) out) -> Tensor(a!)
+- func: gelu.out(Tensor self, int approximate=0, *, Tensor(a!) out) -> Tensor(a!)
```

Comment:
I propose we change the type of `approximate` to be `str` for all overloads in `native_functions.yaml` (note: C++-side, `str` becomes `c10::string_view`).
* This removes the str / int discrepancy between python / C++ and allows us to remove the `torch.nn.functional.gelu` function wrapper. In `torch.nn.functional.py`, the gelu code should once again use the approach of:
```python
gelu = _add_docstr(
    torch._C._nn.gelu,
    r"""
    ...
    """
```
* Now we no longer need the `elif not torch.jit.is_scripting` nonsense because #67565 no longer applies.
* We should be able to make the new arg kw-only again (e.g. `gelu.out(Tensor self, *, str approximate='none', Tensor(a!) out) -> Tensor(a!)`). The test error comes from the *wrapper* accepting kw-only args, so no wrapper -> no problem.
* There's no longer a need for the new `_gelu.py` file.
* Can still keep an enum defined for nice C++-side usage (see what's done for [`scatter.reduce`](https://github.com/pytorch/pytorch/blob/2d4513e4c43aa11860785bdf73c0921588b7fb11/aten/src/ATen/native/TensorAdvancedIndexing.cpp#L77-L85)).